docs/pages/cairo/hello-dojo.md

```diff
@@ -144,19 +143,20 @@ mod actions {
             let (mut position, mut moves) = get!(world, player, (Position, Moves));
 
             // Deduct one from the player's remaining moves.
-            moves.remaining -= 1;
+            let remaining = moves.remaining - 1;
+            moves.remaining = remaining;
 
             // Update the last direction the player moved in.
             moves.last_direction = direction;
 
             // Calculate the player's next position based on the provided direction.
             let next = next_position(position, direction);
 
-            // // Update the world state with the new moves data and position.
+            // Update the world state with the new moves data and position.
             set!(world, (moves, next));
-
             // Emit an event to the world to notify about the player's move.
-            emit!(world, Moved { player, direction });
+            // emit!(world, Moved { player, direction });
+            emit!(world,( Moves { player, remaining, last_direction: direction }));
```

Comment:
Isn't `moves` already containing all this data actually?

Reply:
I fixed to emit event just like this `emit!(world, (moves));`. Fixed on commit [24c5edf](https://github.com/dojoengine/book/pull/254/commits/24c5edf6007a0b2e77d3cb4cd472f43aae2aa51c)